libs/MeshKernel/src/Mesh2D.cpp

```diff
@@ -1934,3 +1936,167 @@ std::vector<int> Mesh2D::NodeMaskFromPolygon(const Polygons& polygon, bool insid
+    regionCount.clear();
+
+    while (!allRegionsLabeled)
+    {
+        UInt elementCount = 0;
+        LabelSingleDomainRegion(regionId, elementRegionId, elementCount);
+
+        if (elementCount > 0)
+        {
+            // If more than 0 elements were labaled, then save this region-id and number of elements.
+            regionCount.push_back({regionId, elementCount});
+            ++regionId;
+        }
+        else
+        {
+            // If no elements were found then we are complete.
+            allRegionsLabeled = true;
+        }
+    }
+}
+
+void Mesh2D::RemoveIslandElements(const UInt regionId, std::vector<UInt>& elementRegionId, UInt& numberOfElementsRemoved)
+{
+    numberOfElementsRemoved = 0;
+
+    // Loop over all element region ids, deleting all those that do not have the same region id as the master id.
+    for (UInt i = 0; i < elementRegionId.size(); ++i)
+    {
+        if (elementRegionId[i] != regionId)
+        {
+            ++numberOfElementsRemoved;
+            std::ranges::for_each(m_facesEdges[i], [this](const UInt edge)
+                                  { DeleteEdge(edge); });
+        }
+    }
+}
+
+void Mesh2D::RemoveIslands()
+{
+    // Label the elements of each discontiguous region of the mesh with a unique identifier for the region.
+    // A mapping between the identifier for the region and the number of elements in the same region is retained
+    // Only the region containing the largest number of elements will be kept, all other regions removed.
+
+    // Map from element id (the domain, index of the array) to region id.
+    std::vector<UInt> elementRegionId;
+    // List of number of elements labeled with number
+    // the domain (index) is the region id and the range, regionCount[id], is the element count.
+    std::vector<std::pair<UInt, UInt>> regionCount;
+
+    // Generate region identifier for all elements in the mesh and a mapping between the region id the the number of elements.
+    LabelAllDomainRegions(elementRegionId, regionCount);
+
+    // If more that 1 region has been found then remove all regions that have fewer elements
+    // than the region with the maximum number of elements.
+    if (regionCount.size() > 1)
+    {
+        // We assume the region with the largest number of elements is the region of interest we would like to retain.
+        // all other regions can (and will) be removed.
+        UInt maxRegionElementCount = 0;
+        UInt mainRegionId = 0;
+
+        for (auto [regionId, regionElementCount] : regionCount)
+        {
+            if (regionElementCount > maxRegionElementCount)
+            {
+                maxRegionElementCount = regionElementCount;
+                mainRegionId = regionId;
+            }
+        }
+
+        UInt numberOfElementsRemoved = 0;
+        // Remove all elements that do not have the main region id.
+        RemoveIslandElements(mainRegionId, elementRegionId, numberOfElementsRemoved);
+        Administrate();
```

Comment:
Administrate() changes the face indices, is this going to be a problem?

Reply:
After a discussion it is ok